util/testkit/testkit.go

```diff
@@ -255,6 +255,17 @@ func (tk *TestKit) MustNoGlobalStats(table string) bool {
 	return true
 }
 
+// MustPartition checks if the result execution plan must read specific partitions.
+func (tk *TestKit) MustPartition(sql string, partitions string, args ...interface{}) bool {
+	rs := tk.MustQuery("explain "+sql, args...)
+	for i := range rs.rows {
+		if strings.Compare(rs.rows[i][3], "partition:"+partitions) == 0 {
```

Comment:
Should we extend this to return which partitions (as an int set/array) will be used? So we can check that a query is only using a specific set of partitions (not more and not less). Or should we have that as a separate helper function later?